core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -0,0 +1,221 @@
+    } else {
+      // avoid returning an empty struct, which is not always supported.
+      // instead, drop the partition field
+      return TypeUtil.selectNot(result, Sets.newHashSet(MetadataColumns.PARTITION_COLUMN_ID));
+    }
+  }
+
+  public static class PositionDeletesBatchScan
+      extends SnapshotScan<BatchScan, ScanTask, ScanTaskGroup<ScanTask>> implements BatchScan {
+
+    protected PositionDeletesBatchScan(Table table, Schema schema) {
+      super(table, schema, new TableScanContext());
+    }
+
+    protected PositionDeletesBatchScan(Table table, Schema schema, TableScanContext context) {
+      super(table, schema, context);
+    }
+
+    @Override
+    protected PositionDeletesBatchScan newRefinedScan(
+        Table newTable, Schema newSchema, TableScanContext newContext) {
+      return new PositionDeletesBatchScan(newTable, newSchema, newContext);
+    }
+
+    @Override
+    public CloseableIterable<ScanTaskGroup<ScanTask>> planTasks() {
+      return TableScanUtil.planTaskGroups(
+          planFiles(), targetSplitSize(), splitLookback(), splitOpenFileCost());
+    }
+
+    @Override
+    protected List<String> scanColumns() {
+      return context().returnColumnStats() ? DELETE_SCAN_WITH_STATS_COLUMNS : DELETE_SCAN_COLUMNS;
+    }
+
+    @Override
+    protected CloseableIterable<ScanTask> doPlanFiles() {
+      String schemaString = SchemaParser.toJson(tableSchema());
+
+      // prepare transformed partition specs and caches
+      Map<Integer, PartitionSpec> transformedSpecs =
+          table().specs().values().stream()
+              .map(spec -> transformSpec(tableSchema(), spec))
+              .collect(Collectors.toMap(PartitionSpec::specId, spec -> spec));
+
+      LoadingCache<Integer, ResidualEvaluator> residualCache =
+          partitionCacheOf(
+              transformedSpecs,
+              spec ->
+                  ResidualEvaluator.of(
+                      spec,
+                      shouldIgnoreResiduals() ? Expressions.alwaysTrue() : filter(),
+                      isCaseSensitive()));
+
+      LoadingCache<Integer, String> specStringCache =
+          partitionCacheOf(transformedSpecs, PartitionSpecParser::toJson);
+
+      LoadingCache<Integer, ManifestEvaluator> evalCache =
+          partitionCacheOf(
+              transformedSpecs,
+              spec -> ManifestEvaluator.forRowFilter(filter(), spec, isCaseSensitive()));
+
+      // iterate through delete manifests
+      CloseableIterable<ManifestFile> deleteManifests =
+          CloseableIterable.withNoopClose(snapshot().deleteManifests(table().io()));
+      CloseableIterable<ManifestFile> filteredManifests =
+          CloseableIterable.filter(
+              deleteManifests,
+              manifest -> evalCache.get(manifest.partitionSpecId()).eval(manifest));
+      Iterable<CloseableIterable<ScanTask>> results =
+          CloseableIterable.transform(
+              filteredManifests,
+              manifest -> {
```

Comment:
Would this logic read all manifests on the driver and suffer from the issue fixed by #5206? I think we did a trick for regular planning to make sure manifests were read in parallel. One way to check is to log thread name before reading and see if different threads read manifests. I'd test this as-is and see if needs changes.

Reply:
You are right, I dont see how this implementation does it on multiple threads.  Changed to follow the code in ManifestGroup